fullstop-violation/fullstop-violation-jpa/src/main/java/org/zalando/stups/fullstop/violation/service/impl/ApplicationVersionServiceImpl.java

```diff
@@ -24,7 +26,12 @@
 
     @Override
     @Transactional(REQUIRES_NEW)
-    public Stack saveStack(final String applicationId, final String applicationVersion) {
+    public Optional<Stack> saveStack(final String applicationId, final String applicationVersion) {
+
+        if (applicationId == null || applicationVersion == null) {
```

Comment:
In some cases, we only have the app id and not the version (see ScmCommitsJob) and we should try to save at least this single information.